src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -204,32 +225,65 @@ private void printAction(Action action, int limit) throws IOException {
     }
   }
 
+  private Action getAction(Digest actionDigest) throws IOException {
+    Action action;
+    try {
+      action = Action.parseFrom(cache.downloadBlob(actionDigest));
+    } catch (IOException e) {
+      throw new IOException("Could not obtain Action from digest.", e);
+    }
+    return action;
+  }
+
+  // Output for print action command.
+  private void printAction(Digest actionDigest, int limit) throws IOException {
+    Action action = getAction(actionDigest);
+
+    Command command;
+    try {
+      command = Command.parseFrom(cache.downloadBlob(action.getCommandDigest()));
+    } catch (IOException e) {
+      throw new IOException("Could not obtain Command from digest.", e);
+    }
+    System.out.printf("Command [digest: %s]:\n", digestUtil.toString(action.getCommandDigest()));
+    printCommand(command);
+
+    Tree tree = cache.getTree(action.getInputRootDigest());
+    System.out.printf(
+        "\nInput files [total: %d, root Directory digest: %s]:\n",
+        getNumFiles(tree), digestUtil.toString(action.getInputRootDigest()));
+    listTree(Paths.get(""), tree, limit);
+
+    System.out.println("\nOutput files:");
+    printList(command.getOutputFilesList(), limit);
+
+    System.out.println("\nOutput directories:");
+    printList(command.getOutputDirectoriesList(), limit);
+
+    System.out.println("\nPlatform:");
+    if (command.hasPlatform() && !command.getPlatform().getPropertiesList().isEmpty()) {
+      System.out.println(command.getPlatform().toString());
+    } else {
+      System.out.println("(none)");
+    }
+  }
+
   // Display output file (either digest or raw bytes).
-  private void printOutputFile(OutputFile file, boolean showRawOutputs) {
+  private void printOutputFile(OutputFile file) {
     String contentString;
     if (file.hasDigest()) {
       contentString = "Content digest: " + digestUtil.toString(file.getDigest());
-    } else if (showRawOutputs) {
-      contentString =
-          String.format(
-              "Raw contents: '%s', size (bytes): %d",
-              file.getContent().toStringUtf8(), file.getContent().size());
     } else {
-      contentString = "Raw contents (not printed)";
+      contentString = "No digest included.";
```

Comment:
This is a very major server error if that happens. We should make this fact explicit.